e2e/wdio/core/tests/grid-list.e2e-spec.ts

```diff
@@ -0,0 +1,122 @@
+    getElementArrayLength, getText,
+    refreshPage,
+    scrollIntoView, waitForClickable, waitForElDisplayed
+} from '../../driver/wdio';
+
+import { text, productTitle, textLocked, warningColor, successColor, neutralColor, errorColor, color, backGroundColor,
+    fontWeight, bold, classAttribute, isSelected } from '../fixtures/appData/grid-list-content';
+
+describe('Grid-list test suite', function() {
+    const gridListPage: GridListPo = new GridListPo();
+    const {
+        layoutPattern, singleSelectItems, multiSelectModeCheckboxes,
+        moreButton, moreButtonItems, footer, gridListItemsByMode, deleteModeTitle, deleteItemButton, unreadStateItem,
+        errorStateItem, lockedStateItemButton, lockedStateItemText, gridListsArray, gridListsTitle,
+        multiSelectModeSelectedItems, errorStatusIndicator, warningStatusIndicator, neutralStatusIndicator,
+        singleSelectItemsSelected, successStatusIndicator, dragAndDropItems
+    } = gridListPage;
+
+    beforeAll(() => {
+        gridListPage.open();
+    }, 1);
+
+    afterEach(() => {
+        refreshPage();
+        waitForElDisplayed(layoutPattern);
+    }, 1);
+
+    it('Verify clicking on read-more button', () => {
+        let defaultItemsQuantity = 5;
+        for (let i = 0; i < 8; i++) {
+            scrollIntoView(moreButton);
+            click(moreButton);
+            expect(getText(moreButton)).toContain(`${defaultItemsQuantity + 5} / 50`);
+            expect(getElementArrayLength(moreButtonItems)).toEqual(defaultItemsQuantity + 5);
+            defaultItemsQuantity += 5;
+        }
+        click(moreButton);
+        expect(moreButton).not.toBeDisplayed();
+    });
+
+    it('Footer should be displayed and contain information', () => {
+        expect(getText(footer)).toBe(text);
+    });
+
+    it('Verify each grid list contains product counter -> product counter should be displayed for all lists', () => {
+        const arr = getElementArrayLength(gridListsArray);
+        for (let i = 0; i < arr; i++) {
+            expect(getText(gridListsTitle, i)).toContain(productTitle);
+        }
+    });
+
+    it('Verify grid list contains product counter', () => {
+        let productsQuantityFromTitle = getText(deleteModeTitle).replace(/\D/g, '');
+        const itemsArray = elementArray(gridListItemsByMode('delete'));
+        expect(productsQuantityFromTitle).toEqual(itemsArray.length.toString());
+        for (let i = 0; i < itemsArray.length; i++) {
+            scrollIntoView(deleteItemButton);
+            click(deleteItemButton);
+            productsQuantityFromTitle = getText(deleteModeTitle).replace(/\D/g, '');
+            const newArray = elementArray(gridListItemsByMode('delete'));
+            expect(productsQuantityFromTitle).toEqual(newArray.length.toString());
+        }
+    });
+
+    fit(`Verify states: Text should be in bold if item is on unread state, Error message should be displayed in footer if item is on 'error' state
```

Comment:
@nikvalor `fit`

Reply:
@valorkin removed. Will add tslint rule for it